intake/views.py

```diff
@@ -2,118 +2,213 @@
 
         return redirect(redirect_path)
     else:
-        return render(request, 'intake/approval.html', {})
+        context = {
+            'request_id': request_id
+        }
+        return render(request, 'intake/approval.html', context)
 
 @login_required
-def contact(request):
+def contact(request, request_id):
     if request.method == 'POST':
-        return redirect('/urgency')
+
+        update_request = UpdateRequest(request_id, request.POST)
+        update_request.perform()
+
+        return redirect('/requests/' + str(request_id) + '/urgency')
     else:
-        return render(request, 'intake/contact.html', {})
+        context = {
+            'request_id': request_id
+        }
+        return render(request, 'intake/contact.html', context)
 
 @login_required
-def no_approval(request):
-    return render(request, 'intake/no_approval.html', {})
+def no_approval(request, request_id):
+    context = {
+        'request_id': request_id
+    }
+    return render(request, 'intake/no_approval.html', context)
 
 @login_required
-def urgency(request):
+def urgency(request, request_id):
     if request.method == 'POST':
         is_urgent = request.POST.get('urgency', None)
 
         redirect_path = {
-            'true': '/urgency_description',
-            'false': '/description'
+            'true': '/requests/' + str(request_id) + '/urgency_description',
+            'false': '/requests/' + str(request_id) + '/description'
         }.get(is_urgent)
 
         return redirect(redirect_path)
     else:
-        return render(request, 'intake/urgency.html', {})
+        context = {
+            'request_id': request_id
+        }
+        return render(request, 'intake/urgency.html', context)
+
+@login_required
+def urgency_description(request, request_id):
+    if request.method == 'POST':
+
+        update_request = UpdateRequest(request_id, request.POST)
+        update_request.perform()
+
+        return redirect('/requests/' + str(request_id) + '/description')
+    else:
+        context = {
+            'request_id': request_id
+        }
+        return render(request, 'intake/urgency_description.html', context)
 
 @login_required
-def urgency_description(request):
+def description(request, request_id):
     if request.method == 'POST':
-        return redirect('/description')
+
+        update_request = UpdateRequest(request_id, request.POST)
+        update_request.perform()
+
+        return redirect('/requests/' + str(request_id) + '/submit_request')
     else:
-        return render(request, 'intake/urgency_description.html', {})
+        context = {
+            'request_id': request_id
+        }
+        return render(request, 'intake/description.html', context)
 
 @login_required
-def description(request):
+def submit_request(request, request_id):
     if request.method == 'POST':
-        return redirect('/submit_request')
+        attributes = {
+            'submitted_at': datetime.datetime.now()
```

Comment:
If somebody returns and updates a request, would this change the submission time?